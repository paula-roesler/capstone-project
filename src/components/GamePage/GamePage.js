import { Link } from 'react-router-dom'
import { useState } from 'react'
import styled from 'styled-components/macro'
import Player from '../Player'
import Button from '../Button'
import ShowWinner from '../ShowWinner'

export default function GamePage({ players, onScore, resetPlayers, winners }) {
  const [visible, setVisible] = useState('whilePlaying')

  return (
    <>
      {visible === 'whilePlaying' && (
        <WrapperGamePage>
          <h1>Score!</h1>
          {players.map(({ name, score }, index) => (
            <Player
              key={index}
              player={name}
              score={score}
              onScore={() => onScore(index)}
              disabled={players.length <= 1}
            />
          ))}
          <Button as={Link} to="/" onClick={onReset}>
            End game
          </Button>

          <Button
            visible={visible}
            onClick={() => {
              setVisible('winner')
            }}
          >
            Show winner!
          </Button>
        </WrapperGamePage>
      )}

      {visible === 'winner' && (
        <>
          <ShowWinner
            hidden={visible}
            title={'Winner'}
            players={players}
            resetPlayers={resetPlayers}
          />
          <Button as={Link} to="/" onClick={onReset}>
            Play again
          </Button>
        </>
      )}
    </>
  )
}

export const WrapperGamePage = styled.div`
  display: grid;
  gap: 10px;
  padding: 10px;
  text-align: center;
  color: var(--primary);
`
