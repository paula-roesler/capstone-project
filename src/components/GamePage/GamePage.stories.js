import GamePage from './GamePage'

export default {
  title: 'GamePage',
  component: 'GamePage',
}

export const templateGamePage = () => <GamePage />
