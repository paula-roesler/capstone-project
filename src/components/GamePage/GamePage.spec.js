import { screen, render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'

import GamePage from './GamePage'

describe('GamePage', () => {
  it('renders all added players with score button', () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/playing']}>
        <GamePage
          players={[
            { name: 'Sue', score: 10 },
            { name: 'Joe', score: 14 },
          ]}
        />
      </MemoryRouter>
    )

    expect(screen.getByText('Sue')).toBeInTheDocument()
    expect(screen.getByText('Joe')).toBeInTheDocument()
    expect(screen.getByText('10')).toBeInTheDocument()
    expect(screen.getByText('End game')).toBeInTheDocument()
    expect(screen.getByText('Show winner!')).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  screen.debug()
})
