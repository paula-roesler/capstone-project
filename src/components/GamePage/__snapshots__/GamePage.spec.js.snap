// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GamePage renders all added players with score button 1`] = `
<div
  class="GamePage__WrapperGamePage-qqemlq-0 dbpAyY"
>
  <h1>
    Score!
  </h1>
  <section
    class="CountScore"
  >
    <div
      class="Player__SinglePlayer-sc-18wz419-0 ijlHBo"
    >
      <div
        class="Player__SinglePlayerName-sc-18wz419-1 bTAKsq"
      >
        Sue
      </div>
      <button
        class="Button-sc-19qr0ud-0 Player__ScoreButton-sc-18wz419-2 gWKtP iCiGMd"
      >
        10
      </button>
    </div>
  </section>
  <section
    class="CountScore"
  >
    <div
      class="Player__SinglePlayer-sc-18wz419-0 ijlHBo"
    >
      <div
        class="Player__SinglePlayerName-sc-18wz419-1 bTAKsq"
      >
        Joe
      </div>
      <button
        class="Button-sc-19qr0ud-0 Player__ScoreButton-sc-18wz419-2 gWKtP iCiGMd"
      >
        14
      </button>
    </div>
  </section>
  <a
    class="Button-sc-19qr0ud-0 gWKtP"
    href="/"
  >
    End game
  </a>
  <button
    class="Button-sc-19qr0ud-0 gWKtP"
  >
    Show winner!
  </button>
</div>
`;
